File: windows/Liferay.SDK/Service/V73/Cpdefinitioninventory/CpdefinitioninventoryService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.Cpdefinitioninventory
{
	public class CpdefinitioninventoryService : ServiceBase
	{
		public CpdefinitioninventoryService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> AddCpDefinitionInventoryAsync(long userId, long cpDefinitionId, string cpDefinitionInventoryEngine, string lowStockActivity, bool displayAvailability, bool displayStockQuantity, int minStockQuantity, bool backOrders, int minOrderQuantity, int maxOrderQuantity, string allowedOrderQuantities, int multipleOrderQuantity)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);
			_parameters.Add("cpDefinitionId", cpDefinitionId);
			_parameters.Add("cpDefinitionInventoryEngine", cpDefinitionInventoryEngine);
			_parameters.Add("lowStockActivity", lowStockActivity);
			_parameters.Add("displayAvailability", displayAvailability);
			_parameters.Add("displayStockQuantity", displayStockQuantity);
			_parameters.Add("minStockQuantity", minStockQuantity);
			_parameters.Add("backOrders", backOrders);
			_parameters.Add("minOrderQuantity", minOrderQuantity);
			_parameters.Add("maxOrderQuantity", maxOrderQuantity);
			_parameters.Add("allowedOrderQuantities", allowedOrderQuantities);
			_parameters.Add("multipleOrderQuantity", multipleOrderQuantity);

			var _command = new JsonObject()
			{
				{ "/commerce.cpdefinitioninventory/add-cp-definition-inventory", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task DeleteCpDefinitionInventoryAsync(long cpDefinitionInventoryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("cpDefinitionInventoryId", cpDefinitionInventoryId);

			var _command = new JsonObject()
			{
				{ "/commerce.cpdefinitioninventory/delete-cp-definition-inventory", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task<dynamic> FetchCpDefinitionInventoryByCpDefinitionIdAsync(long cpDefinitionId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("cpDefinitionId", cpDefinitionId);

			var _command = new JsonObject()
			{
				{ "/commerce.cpdefinitioninventory/fetch-cp-definition-inventory-by-cp-definition-id", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateCpDefinitionInventoryAsync(long cpDefinitionInventoryId, string cpDefinitionInventoryEngine, string lowStockActivity, bool displayAvailability, bool displayStockQuantity, int minStockQuantity, bool backOrders, int minOrderQuantity, int maxOrderQuantity, string allowedOrderQuantities, int multipleOrderQuantity)
		{
			var _parameters = new JsonObject();

			_parameters.Add("cpDefinitionInventoryId", cpDefinitionInventoryId);
			_parameters.Add("cpDefinitionInventoryEngine", cpDefinitionInventoryEngine);
			_parameters.Add("lowStockActivity", lowStockActivity);
			_parameters.Add("displayAvailability", displayAvailability);
			_parameters.Add("displayStockQuantity", displayStockQuantity);
			_parameters.Add("minStockQuantity", minStockQuantity);
			_parameters.Add("backOrders", backOrders);
			_parameters.Add("minOrderQuantity", minOrderQuantity);
			_parameters.Add("maxOrderQuantity", maxOrderQuantity);
			_parameters.Add("allowedOrderQuantities", allowedOrderQuantities);
			_parameters.Add("multipleOrderQuantity", multipleOrderQuantity);

			var _command = new JsonObject()
			{
				{ "/commerce.cpdefinitioninventory/update-cp-definition-inventory", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateCpDefinitionInventoryAsync(long groupId, long cpDefinitionInventoryId, string cpDefinitionInventoryEngine, string lowStockActivity, bool displayAvailability, bool displayStockQuantity, int minStockQuantity, bool backOrders, int minOrderQuantity, int maxOrderQuantity, string allowedOrderQuantities, int multipleOrderQuantity)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("cpDefinitionInventoryId", cpDefinitionInventoryId);
			_parameters.Add("cpDefinitionInventoryEngine", cpDefinitionInventoryEngine);
			_parameters.Add("lowStockActivity", lowStockActivity);
			_parameters.Add("displayAvailability", displayAvailability);
			_parameters.Add("displayStockQuantity", displayStockQuantity);
			_parameters.Add("minStockQuantity", minStockQuantity);
			_parameters.Add("backOrders", backOrders);
			_parameters.Add("minOrderQuantity", minOrderQuantity);
			_parameters.Add("maxOrderQuantity", maxOrderQuantity);
			_parameters.Add("allowedOrderQuantities", allowedOrderQuantities);
			_parameters.Add("multipleOrderQuantity", multipleOrderQuantity);

			var _command = new JsonObject()
			{
				{ "/commerce.cpdefinitioninventory/update-cp-definition-inventory", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}
	}
}

File: android/services-v73/src/main/java/com/liferay/mobile/android/v73/cpdefinitioninventory/CpdefinitioninventoryService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v73.cpdefinitioninventory;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class CpdefinitioninventoryService extends BaseService {

	public CpdefinitioninventoryService(Session session) {
		super(session);
	}

	public JSONObject addCpDefinitionInventory(long userId, long cpDefinitionId, String cpDefinitionInventoryEngine, String lowStockActivity, boolean displayAvailability, boolean displayStockQuantity, int minStockQuantity, boolean backOrders, int minOrderQuantity, int maxOrderQuantity, String allowedOrderQuantities, int multipleOrderQuantity) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("allowedOrderQuantities", checkNull(allowedOrderQuantities));
			_params.put("backOrders", backOrders);
			_params.put("cpDefinitionId", cpDefinitionId);
			_params.put("cpDefinitionInventoryEngine", checkNull(cpDefinitionInventoryEngine));
			_params.put("displayAvailability", displayAvailability);
			_params.put("displayStockQuantity", displayStockQuantity);
			_params.put("lowStockActivity", checkNull(lowStockActivity));
			_params.put("maxOrderQuantity", maxOrderQuantity);
			_params.put("minOrderQuantity", minOrderQuantity);
			_params.put("minStockQuantity", minStockQuantity);
			_params.put("multipleOrderQuantity", multipleOrderQuantity);
			_params.put("userId", userId);

			_command.put("/commerce.cpdefinitioninventory/add-cp-definition-inventory", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public void deleteCpDefinitionInventory(long cpDefinitionInventoryId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("cpDefinitionInventoryId", cpDefinitionInventoryId);

			_command.put("/commerce.cpdefinitioninventory/delete-cp-definition-inventory", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

	public JSONObject fetchCpDefinitionInventoryByCpDefinitionId(long cpDefinitionId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("cpDefinitionId", cpDefinitionId);

			_command.put("/commerce.cpdefinitioninventory/fetch-cp-definition-inventory-by-cp-definition-id", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateCpDefinitionInventory(long groupId, long cpDefinitionInventoryId, String cpDefinitionInventoryEngine, String lowStockActivity, boolean displayAvailability, boolean displayStockQuantity, int minStockQuantity, boolean backOrders, int minOrderQuantity, int maxOrderQuantity, String allowedOrderQuantities, int multipleOrderQuantity) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("allowedOrderQuantities", checkNull(allowedOrderQuantities));
			_params.put("backOrders", backOrders);
			_params.put("cpDefinitionInventoryEngine", checkNull(cpDefinitionInventoryEngine));
			_params.put("cpDefinitionInventoryId", cpDefinitionInventoryId);
			_params.put("displayAvailability", displayAvailability);
			_params.put("displayStockQuantity", displayStockQuantity);
			_params.put("groupId", groupId);
			_params.put("lowStockActivity", checkNull(lowStockActivity));
			_params.put("maxOrderQuantity", maxOrderQuantity);
			_params.put("minOrderQuantity", minOrderQuantity);
			_params.put("minStockQuantity", minStockQuantity);
			_params.put("multipleOrderQuantity", multipleOrderQuantity);

			_command.put("/commerce.cpdefinitioninventory/update-cp-definition-inventory", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateCpDefinitionInventory(long cpDefinitionInventoryId, String cpDefinitionInventoryEngine, String lowStockActivity, boolean displayAvailability, boolean displayStockQuantity, int minStockQuantity, boolean backOrders, int minOrderQuantity, int maxOrderQuantity, String allowedOrderQuantities, int multipleOrderQuantity) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("allowedOrderQuantities", checkNull(allowedOrderQuantities));
			_params.put("backOrders", backOrders);
			_params.put("cpDefinitionInventoryEngine", checkNull(cpDefinitionInventoryEngine));
			_params.put("cpDefinitionInventoryId", cpDefinitionInventoryId);
			_params.put("displayAvailability", displayAvailability);
			_params.put("displayStockQuantity", displayStockQuantity);
			_params.put("lowStockActivity", checkNull(lowStockActivity));
			_params.put("maxOrderQuantity", maxOrderQuantity);
			_params.put("minOrderQuantity", minOrderQuantity);
			_params.put("minStockQuantity", minStockQuantity);
			_params.put("multipleOrderQuantity", multipleOrderQuantity);

			_command.put("/commerce.cpdefinitioninventory/update-cp-definition-inventory", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

}